tinygrad/engine/jit.py

```diff
@@ -115,7 +115,7 @@ def add_buffer(self, b:Buffer) -> Buffer:
     if found:=self.buffer_replace.get(b, None): return found
     if b.is_allocated() or b.lb_refcount > 0: return b
     if b._base is not None:
-      self.buffer_replace[b] = ret = Buffer(b.device, b.size, b.dtype, base=self.buffer_replace.get(b._base, b._base), offset=b.offset)
+      self.buffer_replace[b] = ret = Buffer(b.device, b.size, b.dtype, base=self.add_buffer(b._base), offset=b.offset)
```

Comment:
With new memory scheduler caught this bug, since a subbuffer can appear earlier than it's base buffer (so this makes the subbuf to point to the wrong old buffer).